6.5.group_by.py

```diff
@@ -0,0 +1,14 @@
+# group_by 6.5.1
+from collections import defaultdict
+
+
+def group_by(func, lst):
+    d = defaultdict(list)
```

Comment:
I know that in this code, 'd' is clear, in a large code it could mean so much, 'Derivative', 'Distance' you have to practice giving meaningfull names,
Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing